### ISSUE TITLE
MedNISTDataset deprecation warning : TypedStorage is deprecated

### ISSUE THREAD

**Issue by rcremese** [Project-MONAI/MONAI#6480]
**Describe the bug**
A deprecation warning is constantly shown when iterating through a dataloader based on medNISTDataset. It surely comes from a dependencie of CachedDaset to [PyTorch 2.0 TypedStorage](https://pytorch.org/docs/stable/storage.html?highlight=typedstorage#torch.TypedStorage).

**To Reproduce**
1. create and activate a virtual env 
2. pip install torch==2.0.0 monai==1.1.0
3. Run the following script
```python
from pathlib import Path
from monai.apps import MedNISTDataset
from monai.data import DataLoader

data_path = Path.home().resolve(strict=True)
dataset = MedNISTDataset(data_path, section='validation', num_workers=1, download=True )
dataloader = DataLoader(dataset, batch_size=10, num_workers=3)
for batch in dataloader:
    print(batch['image'].shape, batch['label'].shape)
```
4. Remove the downloaded dataset from your home folder.

**Expected behavior**
Output the shape of image and label tensors 
i.e : torch.Size([10, 64, 64]) torch.Size([10])
....
Instead, I get a lot of deprecation warning that pollute the console log : `TypedStorage is deprecated. It will be removed in the future and UntypedStorage will be the only storage class. This should only matter to you if you are using storages directly.  To access UntypedStorage directly, use tensor.untyped_storage() instead of tensor.storage()` 

**Environment**

Ensuring you use the relevant python executable, please paste the output of:

```shell
python -c 'import monai; monai.config.print_debug_info()'
================================
Printing MONAI config...
================================
MONAI version: 1.1.0
Numpy version: 1.24.1
Pytorch version: 2.0.0+cu117
MONAI flags: HAS_EXT = False, USE_COMPILED = False, USE_META_DICT = False
MONAI rev id: a2ec3752f54bfc3b40e7952234fbeb5452ed63e3
MONAI __file__: /usr/local/lib/python3.10/dist-packages/monai/__init__.py

Optional dependencies:
Pytorch Ignite version: NOT INSTALLED or UNKNOWN VERSION.
Nibabel version: 5.1.0
scikit-image version: NOT INSTALLED or UNKNOWN VERSION.
Pillow version: 9.5.0
Tensorboard version: 2.11.2
gdown version: NOT INSTALLED or UNKNOWN VERSION.
TorchVision version: NOT INSTALLED or UNKNOWN VERSION.
tqdm version: 4.64.1
lmdb version: NOT INSTALLED or UNKNOWN VERSION.
psutil version: 5.9.4
pandas version: 1.5.3
einops version: 0.6.1
transformers version: NOT INSTALLED or UNKNOWN VERSION.
mlflow version: NOT INSTALLED or UNKNOWN VERSION.
pynrrd version: NOT INSTALLED or UNKNOWN VERSION.

For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies


================================
Printing system config...
================================
System: Linux
Linux version: Ubuntu 20.04.5 LTS
Platform: Linux-4.15.0-159-generic-x86_64-with-glibc2.31
Processor: x86_64
Machine: x86_64
Python version: 3.10.11
Process name: python
Command: ['python', '-c', 'import monai; monai.config.print_debug_info()']
Open files: [popenfile(path='/root/.vscode-server/data/logs/20230505T084445/remoteptyhost.log', fd=19, position=271, mode='a', flags=33793), popenfile(path='/root/.vscode-server/data/logs/20230505T084445/remoteagent.log', fd=20, position=7266, mode='a', flags=33793)]
Num physical CPUs: 20
Num logical CPUs: 40
Num usable CPUs: 40
CPU usage (%): [4.8, 47.6, 4.8, 5.2, 5.2, 7.3, 5.2, 6.0, 5.2, 6.0, 5.6, 57.6, 7.9, 6.4, 6.0, 5.2, 5.2, 11.5, 10.1, 5.2, 5.2, 4.4, 4.8, 4.4, 5.2, 4.8, 5.2, 4.8, 5.2, 4.8, 5.2, 5.2, 4.8, 4.8, 4.8, 4.8, 5.2, 9.6, 4.8, 5.2]
CPU freq. (MHz): 1333
Load avg. in last 1, 5, 15 mins (%): [1.2, 0.8, 0.9]
Disk usage (%): 96.7
Avg. sensor temp. (Celsius): UNKNOWN for given OS
Total physical memory (GB): 251.8
Available memory (GB): 239.3
Used memory (GB): 10.2

================================
Printing GPU config...
================================
Num GPUs: 4
Has CUDA: True
CUDA version: 11.7
cuDNN enabled: True
cuDNN version: 8500
Current device: 0
Library compiled for CUDA architectures: ['sm_37', 'sm_50', 'sm_60', 'sm_70', 'sm_75', 'sm_80', 'sm_86']
GPU 0 Name: Tesla V100-DGXS-32GB
GPU 0 Is integrated: False
GPU 0 Is multi GPU board: False
GPU 0 Multi processor count: 80
GPU 0 Total memory (GB): 31.7
GPU 0 CUDA capability (maj.min): 7.0
GPU 1 Name: Tesla V100-DGXS-32GB
GPU 1 Is integrated: False
GPU 1 Is multi GPU board: False
GPU 1 Multi processor count: 80
GPU 1 Total memory (GB): 31.7
GPU 1 CUDA capability (maj.min): 7.0
GPU 2 Name: Tesla V100-DGXS-32GB
GPU 2 Is integrated: False
GPU 2 Is multi GPU board: False
GPU 2 Multi processor count: 80
GPU 2 Total memory (GB): 31.7
GPU 2 CUDA capability (maj.min): 7.0
GPU 3 Name: Tesla V100-DGXS-32GB
GPU 3 Is integrated: False
GPU 3 Is multi GPU board: False
GPU 3 Multi processor count: 80
GPU 3 Total memory (GB): 31.7
GPU 3 CUDA capability (maj.min): 7.0
```

**Additional context**

**Comment by wyli** [comment]
thanks for reporting, it's been addressed in the latest versions, for example monai==1.2.0rc5 https://github.com/Project-MONAI/MONAI/issues/6104
